src/execution_engine.rs

```diff
@@ -158,20 +160,73 @@ impl ExecutionEngine {
         Ok(())
     }
 
-    /// WARNING: The returned address *will* be invalid if the EE drops first
-    /// Do not attempt to transmute it to a function if the ExecutionEngine is gone
-    // TODOC: Initializing a target MUST occur before creating the EE or else it will not count
-    // TODOC: Can still add functions after EE has been created
-    pub fn get_function_address(&self, fn_name: &str) -> Result<u64, FunctionLookupError> {
+    /// Try to load a function from the execution engine.
+    /// 
+    /// If a target hasn't already been initialized, spurious "function not 
+    /// found" errors may be encountered.
+    /// 
+    /// The [`UnsafeFunctionPointer`] trait is designed so only `unsafe extern 
+    /// "C"` functions can be retrieved via the `get_function()` method. If you
+    /// get funny type errors then it's probably because you have specified the
+    /// wrong calling convention or forgotten to specify the retrieved function
+    /// is `unsafe`.
+    /// 
+    /// # Examples
+    /// 
+    /// 
+    /// ```rust
+    /// # use inkwell::targets::{InitializationConfig, Target};
+    /// # use inkwell::context::Context;
+    /// # use inkwell::OptimizationLevel;
+    /// # Target::initialize_native(&InitializationConfig::default()).unwrap();
+    /// let context = Context::create();
+    /// let module = context.create_module("test");
+    /// let builder = context.create_builder();
+    ///
+    /// // Set up the function signature
+    /// let double = context.f64_type();
+    /// let sig = double.fn_type(&[], false);
+    ///
+    /// // Add the function to our module
+    /// let f = module.add_function("test_fn", &sig, None);
+    /// let b = context.append_basic_block(&f, "entry");
+    /// builder.position_at_end(&b);
+    ///
+    /// // Insert a return statement
+    /// let ret = double.const_float(64.0);
+    /// builder.build_return(Some(&ret));
+    ///
+    /// // create the JIT engine
+    /// let mut ee = module.create_jit_execution_engine(OptimizationLevel::None).unwrap();
+    /// 
+    /// // fetch our JIT'd function and execute it
+    /// unsafe {
+    ///     let test_fn = ee.get_function::<unsafe extern "C" fn() -> f64>("test_fn").unwrap();
+    ///     let return_value = test_fn();
+    ///     assert_eq!(return_value, 64.0);
```

Comment:
nit: `assert_eq` doesn't need to be in unsafe block. Maybe return it? IE
```rust
let return_value = unsafe {
    let test_fn = ee.get_function::<unsafe extern "C" fn() -> f64>("test_fn").unwrap();
    test_fn()
};
assert_eq!(return_value, 64.0);
```

Unless you think it's less clear that way, feel free to leave it as is.

Reply:
It doesn't really need to be inside the `unsafe` block, but the reason I wrote it that way to show that the `test_fn` and using anything we get from it is innately `unsafe`, and requires a bit of extra attention.